scripts/hca-utils/hca_utils/utils.py

```diff
@@ -0,0 +1,332 @@
+        blob = bucket.blob(filename)
+        blob.upload_from_filename(filename)
+
+        filepath = f"gs://{self.bucket}/{filename}"
+        print(f"Put a soft-delete file here: {filepath}")
+
+        return filepath
+
+    # jade interactions
+    @staticmethod
+    @lru_cache()
+    def _get_endpoint(endpoint: str):
+        """
+        Get relevant information, such as the HTTP method, the path, and parameters, for Jade API endpoints.
+        :param endpoint: The endpoint name to get information for.
+        :return: A dictionary containing the endpoint's method, path, and parameters.
+        """
+        url = "https://jade-terra.datarepo-prod.broadinstitute.org/v2/api-docs"
+        response = CachedSession().get(url=url)
+        paths = response.json()["paths"]
+
+        my_map = {}
+
+        for path, methods in paths.items():
+            for method, rest in methods.items():
+                if "parameters" in rest:
+                    params = [(param["name"], param["in"]) for param in rest["parameters"]]
+                else:
+                    params = []
+                my_map.update({rest["operationId"]: {"method": method.upper(), "path": path, "params": params}})
+        if endpoint in my_map:
+            return my_map[endpoint]
+        else:
+            raise KeyError(f"Endpoint named {endpoint} not found!")
+
+    def _hit_jade(self, endpoint, handle_ok, body=None, params=None, query=None):
+        """
+        A generic function for hitting the Jade API.
+        :param endpoint: The endpoint to interact with.
+        :param handle_ok: A function that handles the response from the server if the response is a 2xx code.
+        :param body: An optional body for the API request if needed.
+        :param params: Optional endpoint parameters (not query filters) for the API request if needed.
+        :param query: Optional query parameters for the API request if needed.
+        :return: The result of running handle_ok on the server's response, or an HTTPError.
+        """
+        ep_info = self._get_endpoint(endpoint)
+        url = f"{self.base_url}{ep_info['path']}"
+
+        if params:
+            url = url.format(**params)
+        if query:
+            url = f"{url}/?{urllib.parse.urlencode(query)}"
+
+        response = AuthorizedSession(self.jade_creds).request(method=ep_info["method"], url=url, json=body)
+
+        if response.ok:
+            return handle_ok(response)
+        else:
+            raise HTTPError(f"Bad response, got code of: {response.status_code} with response body {response.text}")
+
+    def get_dataset_id(self) -> str:
+        """
+        Get the dataset ID of the provided dataset name.
+        :return: The dataset id.
+        """
+
+        def handle_response(response):
+            return response.json()["items"][0]["id"]
+
+        dataset_id = self._hit_jade("enumerateDatasets", handle_response, query={"filter": self.dataset})
+        return dataset_id
```

Comment:
```suggestion
        jade_response = self._hit_jade("enumerateDatasets", query={"filter": self.dataset})
        return jade_response.json()["items"][0]["id"]
```
example shortened method if we ditch the `handle_ok` callback